src/main/webapp/app/components/tabs/CurationToolsTab.tsx

```diff
@@ -0,0 +1,75 @@
+import { PAGE_ROUTE } from 'app/config/constants/constants';
+import { Gene, Meta } from 'app/shared/model/firebase/firebase.model';
+import GeneSelect from 'app/shared/select/GeneSelect';
+import { getFirebasePath } from 'app/shared/util/firebase/firebase-utils';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+import React, { useEffect, useState } from 'react';
+import { FaExclamationCircle } from 'react-icons/fa';
+import { Button, Col, Row } from 'reactstrap';
+import './curation-tools-tab.scss';
+
+function CurationToolsTab({ addMetaListener, metaData, createMetaGene, createGene }: StoreProps) {
+  const [selectedGene, setSelectedGene] = useState<string>(null);
+  const [createButtonDisabled, setCreateButtonDisabled] = useState(true);
+
+  useEffect(() => {
+    const callback = addMetaListener();
+
+    return () => callback && callback();
+  }, []);
+
+  useEffect(() => {
+    if (selectedGene && !Object.keys(metaData).includes(selectedGene)) {
+      setCreateButtonDisabled(false);
+    } else {
+      setCreateButtonDisabled(true);
+    }
+  }, [selectedGene]);
+
+  async function handleCreateGene() {
+    const gene = new Gene(selectedGene);
+    await Promise.all([
+      createGene(getFirebasePath('GENE', gene.name), gene),
+      createMetaGene(getFirebasePath('META_GENE', gene.name), new Meta()),
+    ]);
```

Comment:
Need to handle the exception when one of the promises failed, we should roll back creation.